laborator/lab-02/README.md

```diff
@@ -0,0 +1,553 @@
+- u este dimensiunea unui element (b pentru 1 octet, h pentru 2, w pentru 4 și g pentru 8 octeți)
+
+## **Pregătire infrastructură**
+
+> **IMPORTANT:** În cadrul laboratoarelor vom folosi repository-ul de git al materiei IOCLA - [https://github.com/systems-cs-pub-ro/iocla](https://github.com/systems-cs-pub-ro/iocla). Repository-ul este clonat pe desktop-ul mașinii virtuale. Pentru a îl actualiza, folosiți comanda `git pull origin master` din interiorul directorului în care se află repository-ul (`~/Desktop/iocla`). Recomandarea este să îl actualizați cât mai frecvent, înainte să începeți lucrul, pentru a vă asigura că aveți versiunea cea mai recentă. Dacă doriți să descărcați repository-ul în altă locație, folosiți comanda `git clone https://github.com/systems-cs-pub-ro/iocla ${target}`. Pentru mai multe informații despre folosirea utilitarului `git`, urmați ghidul de la [Git Immersion](https://gitimmersion.com/).
+
+ Pentru desfășurarea acestui laborator vom folosi interfața în linia de comandă.
+
+ Pe parcursul laboratorului, în linia de comandă, vom folosi:
+
+ - comanda `gcc` pe post de linker
+ - `gdb` pentru analiza dinamică, investigație și debugging
+ 
+ În general nu va fi nevoie să dați comenzi de compilare. Fiecare director cuprinde un Makefile pe care îl puteți rula pentru a compila în mod automat fișierele cod sursă limbaj de asamblare sau C.
+
+ ### **1. Iterarea print-un vector de întregi**
+
+ Veți rezolva exercițiul plecând de la fișierul **ex1.c** aflat în directorul **ex1**.
+
+ Se dă urmatoarea bucată de cod în C:
+
+ ```c
+ #include <stdio.h>
+ 
+int main() {
+    int v[] = {0xCAFEBABE, 0xDEADBEEF, 0x0B00B135, 0xBAADF00D, 0xDEADC0DE};
+ 
+    return 0;
+}
+```
+
+Afișați adresele elementelor din vectorul `v` împreună cu valorile de la acestea. Parcurgeți, pe rând, adresele din `v` octet cu octet, din doi în doi, respectiv din patru în patru octeți.
+
+> **TIP:** Puteți parcurge memoria octet cu octet începând de la o anumită adresă folosind un pointer de tipul `unsigned char*` (pentru că tipul `char` este reprezentat pe un octet).
+>```c
+>unsigned char *char_ptr = &v;
+>```
+> Pentru afișarea adresei, respectiv a valorii puteți folosi:
+>```c
+>printf("%p -> 0x%x\n", char_ptr, *char_ptr);
+>```
+
+### **2. Ștergerea primei apariții a unui pattern dintr-un șir de caractere**
+
+Veți rezolva exercițiul plecând de la fișierul **ex2.c** aflat în directorul **ex2**.
+
+Dându-se un șir de caractere și un pattern să se implementeze funcția `delete_first(char *s, char *pattern)` care întoarce șirul obținut prin ștergerea primei apariții a pattern-ului în s.
+
+> **NOTE:** Pentru `s = "Ana are mere"` și `pattern = "re"` se va returna șirul "Ana a mere".
+
+> **IMPORTANT:**
+> Atenție
+> ```c
+> char* s = "Ana are mere" se alocă în .rodata;
+> char s[] = "Ana are mere" se alocă în .text dacă declarația e într-o funcție;
+> char s[] = "Ana are mere" se alocă în .data dacă declarația e în afara funcției;
```

Comment:
Aici informația pare greșită / incompletă - se poate corecta și la partea de dezvoltare / review.
 - `char *s` - șirul se alocă în `.rodata`, dar pointer-ul este în zona de memorie specifică alocării - stivă / `.data`.
 - `char s[]` se alocă pe stivă în mod normal, dacă este într-o funcție, pentru că este read-write. Chiar și `const char s[]` o să fie tot pe stivă pentru funcții, chiar dacă o să fie în `.rodata` pentru date globale. Este însă vorba de decizii luate de compilator, nu sunt obligatoriu implementate la fel mereu.

Reply:
Mi s-a parut si mie dubios acest paragraf cand l-am citit (mai ales ca se mentioneaza segmentele desi nu cred ca se ajunge cu materia acolo la momentul laboratorului 2), dar am pastrat in aceasta faza textul laboratorului asa cum este pe ocw (doar cu typo-uri corectate)

Reply:
Ești responsabil cu conținutul. Lasă-l așa acum, dar poți să îl corectezi / scoți cum consideri.